src/passes/identifierManglerPass/declarationNameMangler.ts

```diff
@@ -120,8 +121,24 @@ export class DeclarationNameMangler extends ASTMapper {
     return `${MANGLED_WARP}${this.nameCounter++}${existingName !== '' ? `_${existingName}` : ''}`;
   }
 
+  checkCollision(node: VariableDeclaration): boolean {
+    const parentContract = node.getClosestParentByType(ContractDefinition);
+    const parentScope = node.vScope;
+
+    if (parentContract !== undefined) {
+      return (
+        parentContract.name === node.name ||
+        parentScope
+          .getChildrenByType(Identifier, true)
+          .some((identifier) => identifier.name === node.name)
+      );
+    }
```

Comment:
Imo `if (parentContract === undefined) return false;` would be more straightforward than inequality comparison